command/ca/admin/admin.go

```diff
@@ -0,0 +1,153 @@
+			listCommand(),
+			//getCommand(),
+			addCommand(),
+			removeCommand(),
+			updateCommand(),
+		},
+		Description: `The **step ca admin** command group provides facilities for managing the
+certificate authority admins.
+
+A admin is an entity that manages administrative resources within a certificate
+authority. Admins manage
+
+* certificate authority configuration
+* provisioner configuration
+* other admins and admin privileges
+
+## EXAMPLES
+
+List the active admins:
+'''
+$ step ca admin list
+'''
+
+Add an admin:
+'''
+$ step ca admin add max@smallstep.com my-jwk-provisioner
+'''
+
+Remove an admin:
+'''
+$ step ca admin remove max@smallstep.com my-jwk-provisioner
+'''`,
+	}
+}
+
+type adminSelect struct {
+	Name     string
+	CLIAdmin *cliAdmin
+}
+
+type cliAdmin struct {
+	*linkedca.Admin
+	ProvisionerName string
+	ProvisionerType string
+}
+
+func toCLI(ctx *cli.Context, client *ca.AdminClient, adm *linkedca.Admin) (*cliAdmin, error) {
+	p, err := client.GetProvisioner(ca.WithProvisionerID(adm.ProvisionerId))
+	if err != nil {
+		return nil, err
+	}
+	return &cliAdmin{Admin: adm, ProvisionerName: p.GetName(), ProvisionerType: p.GetType().String()}, nil
+}
+
+func listToCLI(ctx *cli.Context, client *ca.AdminClient, admins []*linkedca.Admin) ([]*cliAdmin, error) {
+	var (
+		err       error
+		cliAdmins = make([]*cliAdmin, len(admins))
+	)
+	for i, adm := range admins {
+		cliAdmins[i], err = toCLI(ctx, client, adm)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return cliAdmins, nil
+}
+
+func adminPrompt(ctx *cli.Context, client *ca.AdminClient, admins []*linkedca.Admin) (*cliAdmin, error) {
+	if len(admins) == 0 {
+		return nil, errors.New("no admins to update")
+	}
+	args := ctx.Args()
+	subject := args[0]
+
+	cliAdmins, err := listToCLI(ctx, client, admins)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter by subject
+	cliAdmins = adminFilter(cliAdmins, func(adm *cliAdmin) bool {
+		return adm.Subject == subject
+	})
+	if len(cliAdmins) == 0 {
+		return nil, fmt.Errorf("no admins with subject %s", subject)
+	}
+
+	// Filter by provisionerName
+	if provName := ctx.String("provisioner"); len(provName) != 0 {
+		cliAdmins = adminFilter(cliAdmins, func(a *cliAdmin) bool {
+			return a.ProvisionerName == provName
+		})
+		if len(cliAdmins) == 0 {
+			return nil, errs.InvalidFlagValue(ctx, "provisioner", provName, "")
+		}
+	}
+
+	// Select admin
+	var items []*adminSelect
+	for _, adm := range cliAdmins {
+		items = append(items, &adminSelect{
+			Name: fmt.Sprintf("subject: %s, provisioner: %s(%s), type: %s", adm.Subject,
+				adm.ProvisionerName, adm.ProvisionerType, adm.Type),
+			CLIAdmin: adm,
+		})
+	}
+
+	if len(items) == 1 {
+		if err := ui.PrintSelected("Admin", items[0].Name); err != nil {
+			return nil, err
+		}
+		return items[0].CLIAdmin, nil
+	}
+
+	i, _, err := ui.Select("Select an admin:", items, ui.WithSelectTemplates(ui.NamedSelectTemplates("Admin")))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This didn't look nice, at least when removing one provisioner:

```sh
$ step-dev beta ca admin list --x5c-cert x5c.crt --x5c-key x5c.key
SUBJECT				PROVISIONER			TYPE
admin@smallstep.com		mariano@smallstep.com (JWK)	ADMIN
super-admin@smallstep.com	mariano@smallstep.com (JWK)	SUPER_ADMIN
mariano@smallstep.com		mariano@smallstep.com (JWK)	SUPER_ADMIN
$ step-dev beta ca admin remove super-admin@smallstep.com --x5c-cert x5c.crt --x5c-key x5c.key
✔ Admin: subject: super-admin@smallstep.com, provisioner: mariano@smallstep.com(JWK), type: SUPER_ADMIN
```

Reply:
What would you prefer?

Reply:
Then it will take up two lines rather than one. Not sure that's a good idea.

Reply:
How do I add that top line using ui.Select? If you can point me to some code that does something similar in the UI prompt I'll copy it.